File: app/src/main/java/starbright/com/projectegg/data/local/model/response/RecipeDetailResponse.java
/**
 * Created by Andreas on 15/8/2018.
 */

package starbright.com.projectegg.data.local.model.response;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class RecipeDetailResponse {

    @SerializedName("id")
    private int mId;

    @SerializedName("title")
    private String mTitle;

    @SerializedName("image")
    private String mImageStringUrl;

    @SerializedName("preparationMinutes")
    private int mPreparationMinutes;

    @SerializedName("cookingMinutes")
    private int mCookingMinutes;

    @SerializedName("sourceUrl")
    private String mSourceStringUrl;

    @SerializedName("sourceName")
    private String mSourceName;

    @SerializedName("analyzedInstructions")
    private InstructionResponse mAnalyzedInstructions;

    @SerializedName("extendedIngredients")
    private List<ExtendedIngredientResponse> mIngredients;

    public int getId() {
        return mId;
    }

    public String getTitle() {
        return mTitle;
    }

    public String getImageStringUrl() {
        return mImageStringUrl;
    }

    public int getPreparationMinutes() {
        return mPreparationMinutes;
    }

    public int getCookingMinutes() {
        return mCookingMinutes;
    }

    public String getSourceStringUrl() {
        return mSourceStringUrl;
    }

    public String getSourceName() {
        return mSourceName;
    }

    public InstructionResponse getAnalyzedInstructions() {
        return mAnalyzedInstructions;
    }

    public List<ExtendedIngredientResponse> getExtendedIngredientResponse() {
        return mIngredients;
    }
}

File: app/src/main/java/starbright/com/projectegg/data/model/response/RecipeDetailResponse.kt
/**
 * Created by Andreas on 15/8/2018.
 */
package starbright.com.projectegg.data.model.response

import com.google.gson.annotations.SerializedName

data class RecipeDetailResponse (
    @SerializedName("id")
    val id: Int = 0,

    @SerializedName("title")
    val title: String,

    @SerializedName("image")
    val imageStringUrl: String? = null,

    @SerializedName("readyInMinutes")
    val cookingTime: Int = 0,

    @SerializedName("sourceUrl")
    val sourceStringUrl: String? = null,

    @SerializedName("sourceName")
    val sourceName: String? = null,

    @SerializedName("servings")
    val servings: Int = 0,

    @SerializedName("nutrition")
    val nutrients: NutritionResponse,

    @SerializedName("dishTypes")
    val dishTypes: List<String>,

    @SerializedName("cuisines")
    val cuisines: List<String>,

    @SerializedName("analyzedInstructions")
    val analyzedInstructions: List<InstructionResponse>,

    @SerializedName("extendedIngredients")
    val extendedIngredientResponse: List<ExtendedIngredientResponse>
)